components/org.wso2.carbon.identity.query.saml/src/test/java/org/wso2/carbon/identity/query/saml/validation/SAMLAuthQueryValidatorTest.java

```diff
@@ -0,0 +1,201 @@
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.wso2.carbon.identity.query.saml.validation;
+
+import org.mockito.Mock;
+import org.opensaml.saml.common.SAMLVersion;
+import org.opensaml.saml.saml2.core.RequestAbstractType;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.testng.PowerMockTestCase;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.carbon.base.CarbonBaseConstants;
+import org.wso2.carbon.identity.core.model.SAMLSSOServiceProviderDO;
+import org.wso2.carbon.identity.query.saml.dto.InvalidItemDTO;
+import org.wso2.carbon.identity.query.saml.exception.IdentitySAML2QueryException;
+import org.wso2.carbon.identity.query.saml.internal.SAMLQueryServiceComponent;
+import org.wso2.carbon.identity.query.saml.util.OpenSAML3Util;
+import org.wso2.carbon.identity.query.saml.util.SAMLQueryRequestConstants;
+import org.wso2.carbon.identity.query.saml.util.SAMLQueryRequestUtil;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.UserRealm;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.service.RealmService;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.testng.Assert.*;
+import static org.wso2.carbon.identity.query.saml.validation.TestUtil.initPrivilegedCarbonContext;
+
+/**
+ * Test Class for the SAMLAuthQueryValidator
+ */
+@PrepareForTest({MultitenantUtils.class, SAMLQueryServiceComponent.class, SAMLQueryRequestUtil.class, OpenSAML3Util.class})
+public class SAMLAuthQueryValidatorTest extends PowerMockTestCase {
+
+    @Mock
+    RealmService testRealmService;
+    @Mock
+    UserRealm testUserRealm;
+    @Mock
+    UserStoreManager testuserStoreManager;
+
+    SAMLAuthQueryValidator testclass = new SAMLAuthQueryValidator();
+    List<InvalidItemDTO> invalidItems = new ArrayList<>();
+
+    @BeforeClass
+    public void setUp() {
+
+        initPrivilegedCarbonContext("testDomain", 1, "testuser");
```

Comment:
Can we support data providers here as well? Then we can test both super-tenant and tenants in a single test.

Reply:
Do we have to test this class for super-tenant and tenants?

Reply:
Yes..